samcli/lib/remote_invoke/sqs_invoke_executors.py

```diff
@@ -0,0 +1,163 @@
+                    LOG.warning(
+                        "MessageBody is defined using the value provided for either --event or --event-file options."
+                    )
+                elif parameter_key == DELAY_SECONDS:
+                    self.request_parameters[parameter_key] = int(parameter_value)
+                elif parameter_key in {MESSAGE_ATTRIBUTES, MESSAGE_SYSTEM_ATTRIBUTES}:
+                    # Load these parameters as dict which is the format required for the send_message boto3 API
+                    # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs/client/send_message.html
+                    self.request_parameters[parameter_key] = json.loads(parameter_value)
+                else:
+                    self.request_parameters[parameter_key] = parameter_value
+        except (ValueError, JSONDecodeError) as err:
+            raise InvalidResourceBotoParameterException(f"Invalid value provided for parameter {parameter_key}", err)
+
+    def _execute_action(self, payload: str) -> RemoteInvokeIterableResponseType:
+        """
+        Calls "send_message" method to send a message to the SQS queue.
+
+        Parameters
+        ----------
+        payload: str
+            The MessageBody which will be sent to the SQS
+
+        Yields
+        ------
+        RemoteInvokeIterableResponseType
+            Response that is consumed by remote invoke consumers after execution
+        """
+        if payload:
+            self.request_parameters[MESSAGE_BODY] = payload
+        else:
+            self.request_parameters[MESSAGE_BODY] = "{}"
+            LOG.debug("Input event not found, sending a message with MessageBody {}")
+        self.request_parameters[QUEUE_URL] = self._queue_url
+        LOG.debug(
+            "Calling sqs_client.send_message with QueueUrl:%s, MessageBody:%s",
+            self.request_parameters[QUEUE_URL],
+            payload,
+        )
+        try:
+            send_message_response = cast(dict, self._sqs_client.send_message(**self.request_parameters))
+
+            if self._remote_output_format == RemoteInvokeOutputFormat.JSON:
+                yield RemoteInvokeResponse(send_message_response)
+            if self._remote_output_format == RemoteInvokeOutputFormat.TEXT:
+                # Create an object with MD5OfMessageBody and MessageId fields, and write to stdout
+                send_message_text_output = SqsSendMessageTextOutput(
+                    MD5OfMessageBody=send_message_response["MD5OfMessageBody"],
+                    MessageId=send_message_response["MessageId"],
+                    MD5OfMessageAttributes=send_message_response.get("MD5OfMessageAttributes"),
+                )
+                output_data = asdict(
+                    send_message_text_output, dict_factory=lambda x: {k: v for (k, v) in x if v is not None}
+                )
+                yield RemoteInvokeResponse(output_data)
+                return
```

Comment:
Do we need a `return` here?

Reply:
Nope `return` is not required here. This was used so that the function doesn't yield anything else but since yield is only called once, `return` can be removed.